client/components/phone-input/test/phone-number__format-number.test.js

```diff
@@ -0,0 +1,220 @@
+
+		{ number: '2', country: 'HU', expected: '2' },
+		{ number: '24', country: 'HU', expected: '24' },
+		{ number: '06243', country: 'HU', expected: '0624 3' },
+		{ number: '0624 32', country: 'HU', expected: '0624 32' },
+		{ number: '0624 321', country: 'HU', expected: '0624 321' },
+		{ number: '0624 3212', country: 'HU', expected: '0624 321 2' },
+		{ number: '0624 321 23', country: 'HU', expected: '0624 321 23' },
+		{ number: '0624 321 234', country: 'HU', expected: '0624 321 234' },
+		{ number: '0624 321 2345', country: 'HU', expected: '0624 321 2345' },
+
+		{ number: '2', country: 'RU', expected: '2' },
+		{ number: '24', country: 'RU', expected: '24' },
+		{ number: '8243', country: 'RU', expected: '8243' },
+		{ number: '82432', country: 'RU', expected: '8243-2' },
+		{ number: '8243-21', country: 'RU', expected: '8243-21' },
+		{ number: '8243-212', country: 'RU', expected: '8243-21-2' },
+		{ number: '8243-21-23', country: 'RU', expected: '8243-21-23' },
+	] )(
+		`Number is formatted as it is typed with formatting included in the input ($number)`,
+		function ( { number, country, expected } ) {
+			const result = formatNumber( number, countries[ country ] );
+			expect( result ).toEqual( expected );
+		}
+	);
+
+	test.each( [
+		{ number: '9876543210', country: 'IS', expected: '9876543210' },
+		{ number: '+96', country: 'IQ', expected: '+96' },
+		{ number: '+126', country: 'AG', expected: '+126' },
+	] )( `Prefix is not added for $country`, function ( { number, country, expected } ) {
+		const result = formatNumber( number, countries[ country ] );
+		expect( result ).toEqual( expected );
+	} );
+} );
+
+describe( 'NANPA', () => {
+	test( 'Formats full length number', function () {
+		const result = formatNumber( '14252222222', countries.US );
+		expect( result ).toEqual( '1 425-222-2222' );
+	} );
+
+	test.each( [
+		{ number: '14', country: 'US', expected: '14' },
+		{ number: '142', country: 'US', expected: '1 42' },
+		{ number: '1425', country: 'US', expected: '1 425' },
+		{ number: '14256', country: 'US', expected: '1 425-6' },
+		{ number: '142565', country: 'US', expected: '1 425-65' },
+		{ number: '1425655', country: 'US', expected: '1 425-655' },
+		{ number: '14256559', country: 'US', expected: '1 425-655-9' },
+		{ number: '142565599', country: 'US', expected: '1 425-655-99' },
+		{ number: '1425655999', country: 'US', expected: '1 425-655-999' },
+		{ number: '14256559999', country: 'US', expected: '1 425-655-9999' },
+	] )( `Number is formatted as it is typed ($number)`, function ( { number, country, expected } ) {
+		const result = formatNumber( number, countries[ country ] );
+		expect( result ).toEqual( expected );
+	} );
+} );
+
+describe( 'Sanitization', () => {
+	test.each( [
+		{ number: '1aaaa', country: 'US', expected: '1' },
+		{ number: '1a', country: 'US', expected: '1' },
+	] )(
+		'Strip non-digits on string of length less than 3',
```

Comment:
Same here -- should we add something from the input to the test name?

Reply:
Fixed in e10751336012e40c886f517b08cc4f2cbe15d2e0.